app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php

```diff
@@ -251,6 +256,25 @@ public function execute(\Magento\Framework\Event\Observer $observer)
     }
 
     /**
+     * @param $jobCode
+     *
+     * @return bool
+     */
+    protected function isJobRunning($jobCode)
+    {
+        $runningJobs = $this->_getRunningSchedules();
```

Comment:
Would be great to replace this part to SQL query, that will be much effective than getting collection and iterate it.